File: app/src/main/java/com/ericho/coupleshare/http/model/BaseSingleResponse.java
package com.ericho.coupleshare.http.model;

import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * Created by steve_000 on 15/4/2017.
 */
public class BaseSingleResponse<T> {
    @SerializedName("status")
    private boolean status;
    @SerializedName("extra")
    private T extra;
    @SerializedName("errorMessage")
    private String errorMessage;
    @SerializedName("otherMessage")
    private String otherMessage;//use when success;

    public boolean isStatus() {
        return status;
    }

    public void setStatus(boolean status) {
        this.status = status;
    }

    public T getExtra() {
        return extra;
    }

    public void setExtra(T extra) {
        this.extra = extra;
    }

    public String getErrorMessage() {
        return errorMessage;
    }

    public void setErrorMessage(String errorMessage) {
        this.errorMessage = errorMessage;
    }

    public String getOtherMessage() {
        return otherMessage;
    }

    public void setOtherMessage(String otherMessage) {
        this.otherMessage = otherMessage;
    }
}

File: app/src/main/java/com/ericho/coupleshare/http/model/BaseSingleResponse.kt
package com.ericho.coupleshare.http.model

import com.google.gson.annotations.SerializedName

/**
 * Created by steve_000 on 7/7/2017.
 * for project CoupleShare
 * package name com.ericho.coupleshare.http.model
 */
class BaseSingleResponse<T> {
    @SerializedName("status")
    var status: Boolean = false
    @SerializedName("extra")
    var extra: T? = null
    @SerializedName("errorMessage")
    var errorMessage: String? = null
    @SerializedName("otherMessage")
    var otherMessage: String? = null//use when success;
}